backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/createbpartner/CreateBPartner_StepDef.java

```diff
@@ -0,0 +1,331 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.cucumber.stepdefs.createbpartner;
+
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.composite.BPartner;
+import de.metas.bpartner.composite.BPartnerComposite;
+import de.metas.bpartner.composite.BPartnerContact;
+import de.metas.bpartner.composite.BPartnerLocation;
+import de.metas.bpartner.composite.repository.BPartnerCompositeRepository;
+import de.metas.common.bpartner.request.JsonRequestBPartner;
+import de.metas.common.bpartner.request.JsonRequestBPartnerUpsert;
+import de.metas.common.bpartner.request.JsonRequestBPartnerUpsertItem;
+import de.metas.common.bpartner.request.JsonRequestComposite;
+import de.metas.common.bpartner.request.JsonRequestContact;
+import de.metas.common.bpartner.request.JsonRequestContactUpsert;
+import de.metas.common.bpartner.request.JsonRequestContactUpsertItem;
+import de.metas.common.bpartner.request.JsonRequestLocation;
+import de.metas.common.bpartner.request.JsonRequestLocationUpsert;
+import de.metas.common.bpartner.request.JsonRequestLocationUpsertItem;
+import de.metas.common.bpartner.response.JsonResponseBPartnerCompositeUpsert;
+import de.metas.common.bpartner.response.JsonResponseBPartnerCompositeUpsertItem;
+import de.metas.common.rest_api.JsonExternalId;
+import de.metas.common.rest_api.JsonMetasfreshId;
+import de.metas.common.rest_api.SyncAdvise;
+import de.metas.cucumber.stepdefs.DataTableUtil;
+import de.metas.cucumber.stepdefs.context.TestContext;
+import de.metas.i18n.Language;
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import lombok.NonNull;
+import org.testcontainers.shaded.com.fasterxml.jackson.core.JsonProcessingException;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class CreateBPartner_StepDef
+{
+
+	private final BPartnerCompositeRepository bPartnerCompositeRepository;
+	private final TestContext testContext;
+	private final JsonRequestComposite.JsonRequestCompositeBuilder jsonRequestCompositeBuilder = JsonRequestComposite.builder();
+
+	public CreateBPartner_StepDef(final TestContext testContext)
+	{
+		this.testContext = testContext;
+		this.bPartnerCompositeRepository = new BPartnerCompositeRepository(new MockLogEntriesRepository());
```

Comment:
```suggestion
		this.bPartnerCompositeRepository = SpringContextHolder.instance.getBean(BPartnerCompositeRepository.class);
```

and then you can ditch the whole MockLogEntriesRepository AFAIU